sklearn_numba_dpex/kmeans/kernels.py

```diff
@@ -0,0 +1,566 @@
+                    ):
+                        item = current_centroids[
+                            global_window_row, global_window_col
+                        ]
+                    else:
+                        item = f32zero
+
+                    window_shm[window_row, window_col] = item
+
+                    load_first_dim_local_idx += cluster_window_per_thread_group
+
+                dpex.barrier()
+
+                for d in range(h):
+                    current_dim_global_idx = d + first_dim_global_idx
+                    if (current_dim_global_idx < dim) and (global_thread < n):
+                        # performance for the line thereafter relies on L1 cache
+                        X_feature = X[current_dim_global_idx, global_thread]
+                    else:
+                        X_feature = f32zero
+                    for i in range(r):
+                        centroid_feature = window_shm[d, i]
+                        partial_scores[i] += centroid_feature * X_feature
+
+                dpex.barrier()
+
+                first_dim_global_idx += h
+
+            for i in range(r):
+                current_score = (
+                    local_centroids_half_l2_norm[i] - partial_scores[i]
+                )
+                if current_score < min_score:
+                    min_score = current_score
+                    min_idx = first_centroid_global_idx + i
+
+            dpex.barrier()
+
+            first_centroid_global_idx += r
+
+        if global_thread >= n:
+            return
+
+        inertia[global_thread] = min_score
+
+        warp_id = global_thread // warp_size
+        private_id = warp_id % nb_centroids_private_copies
+        dpex.atomic.add(
+            centroid_counts_private_copies,
+            (private_id, min_idx),
+            f32one,
+        )
+
+        for d in range(dim):
+            dpex.atomic.add(
+                centroids_private_copies,
+                # ((warp_id + d) % nb_centroids_private_copies, d, min_idx),
+                (private_id, d, min_idx),
+                X[d, global_thread],
+            )
+
+    return (
+        nb_centroids_private_copies,
+        fused_kernel_fixed_window[global_size, work_group_size],
+    )
+
+
+@cached_kernel_factory
+def make_assignment_kernel_fixed_window(
+    n,
+    dim,
+    n_clusters,
+    warp_size,
+    window_length_multiple,
+    cluster_window_per_thread_group,
+    number_of_load_iter,
+):
+    r = warp_size * window_length_multiple
+    work_group_size = r * cluster_window_per_thread_group
+    h = number_of_load_iter * cluster_window_per_thread_group
+
+    n_cluster_groups = math.ceil(n_clusters / r)
+    global_size = (math.ceil(n / work_group_size)) * (work_group_size)
+    n_dim_windows = math.ceil(dim / h)
+    window_shm_shape = (h, r)
+
+    inf = float32(math.inf)
+    f32zero = float32(0.0)
+    f32two = float32(2)
+
+    @dpex.kernel
+    def assigment_kernel_fixed_window(
```

Comment:
There is an "n" missing in "assignment". Also, here again, "fixed_window" would rather come at the beginning of the name.

```suggestion
    def fixed_window_assignment_kernel(
```